### PR TITLE
Updated the Variables extension to render label when available.

### LINKED ISSUE #1217: Add a `isVariableSubstitutedAsLabels` prop
Ref: https://github.com/bigbinary/neeto-form-web/issues/8982

https://abhay-ashokan.neetorecord.com/watch/875c5140-ba76-4172-8fec-19ce971b1506

---

- [x] Add a `isVariableSubstitutedAsLabels` prop (false by default). If this variable is enabled, the variables should be rendered using labels instead of ids. This can be implemented by writing a NodeView to render the Editor content.
- [x] In case of longer labels/questions, we can truncate it to 15 characters and then suffix it with ...

@deepakjosp _a

enhancement _t

### PR DESCRIPTION
**Pull request by deepakjosp**
Fixes #1217 

**Description**
- Renders the variable name while editing. The renderHtml is not changes, so the getHtml will still return the id.

**Checklist**

- [ ] ~I have made corresponding changes to the documentation.~
- [ ] ~I have updated the types definition of modified exports.~
- [x] I have verified the functionality in some of the neeto web-apps.
- [x] I have added the necessary label (patch/minor/major - If package publish
      is required).

**Reviewers**

### CODE REVIEW

**Comment by deepakjosp** [comment]
@AbhayVAshokan  _a please review.

File: src/components/Editor/CustomExtensions/Variable/constants.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export const MAX_VARIABLE_NAME_DISPLAY_LENGTH = 15;
```
Comment (AbhayVAshokan) [inline src/components/Editor/CustomExtensions/Variable/constants.js:1]: Let's increase it to 25. There are a lot of variables that get truncated at 15.

```suggestion
export const MAX_VARIABLE_NAME_DISPLAY_LENGTH = 25;
```

**Review by AbhayVAshokan — APPROVED** [review]
- [ ] ~https://abhay-ashokan.neetorecord.com/watch/80544394-141f-4467-b184-eb7c32db61a1~

@deepakjosp _a

**Comment by deepakjosp** [comment]
> https://github.com/bigbinary/neeto-editor/pull/1228#pullrequestreview-2345616988

@AbhayVAshokan  the mentioned issue is not reproducible. The dropdown has a zIndex of `9999` by default. I'm merging the PR. I'll create an issue fix it if the issue is reproducible.

image

**Pull request merged**